framework/src/userobject/ElementUserObject.C

```diff
@@ -52,3 +52,8 @@ ElementUserObject::ElementUserObject(const InputParameters & parameters) :
     addMooseVariableDependency(coupled_vars[i]);
 }
 
+bool
+ElementUserObject::requestedJacobian(unsigned int var) const
+{
+  return var < _requested_jacobian_flag.size() && _requested_jacobian_flag[var];
```

Comment:
Yeah, if it is a set it is for sure cheaper to _just call_ ```executeJacobian```. I think I'd rather remove this mechanism entirely (I've already written a comment to that effect a few hours ago).